illumio_connector.py

```diff
@@ -472,23 +478,57 @@ def _handle_create_enforcement_boundary(self, param):
     def _handle_get_workloads(self, param):
         action_result = self.add_action_result(ActionResult(dict(param)))
 
-        enforcement_mode = param["enforcement_mode"].lower()
+        enforcement_mode = (
+            param.get("enforcement_mode").lower()
+            if param.get("enforcement_mode")
+            else None
+        )
+        connectivity = (
+            param.get("connectivity").lower() if param.get("connectivity") else None
+        )
+        labels = self.handle_comma_seperated_string(param.get("labels", ""))
+        managed = True
+
         if enforcement_mode not in ENFORCEMENT_MODE_LIST:
             return action_result.set_status(
                 phantom.APP_ERROR,
-                "Please enter a valid value for 'enforcement mode' parameter",
+                "Please enter a valid value for 'enforcement_mode' parameter",
+            )
+
+        if connectivity not in CONNECTIVITY_LIST:
```

Comment:
If we need a dozen lines of code to make this joint `online`/`managed` connectivity param work, maybe it should just be split out into two separate optional booleans.

Reply:
Spliting this out into two booleans is what we thought of initially but didnt go with that workflow as providing managed boolean parameter indicates that we can fetch managed(keeping managed boolean as true) and unmanaged(keeping managed boolean as false) data using the action. But the UI does not support fetching of managed workloads and only possible values from UI are Online, Offline and Unmanaged. **We usually mimic UI while developing actions as testing of all the cases would be possible due to it.**

So we are removing unnecessary initialization statements like **managed=True** and **managed=None**. Now the if else condition would be just for online and offline cases.

Reply:
Don't mirror the UI. Mirror the API. The default for both should be None and they should both be optional search terms.

Reply:
Replacing the "connectivity" parameter with two dropdown parameters: "online" and "managed", which will be None if nothing is selected otherwise "True" or "False".